### DISCUSSION TITLE
Rendering on GitHub

### DISCUSSION THREAD

**Discussion by ZackLarsen** [marimo-team/marimo#2868] [Q&A] [answered]
A very useful part of Jupyter notebooks is that the .ipynb files are rendered directly on GitHub so that when you navigate to the files, you can see the output. Is there a way to do this with Marimo or something on the roadmap to make this happen? Without this functionality, it seems you would have to run the notebook on a server to share with colleagues, whereas you could just share a link to the GitHub page for a Jupyter notebook and they would be able to see it.

I realize this may mean losing some of the reactive/interactive features of Marimo, but I am curious if there is a way to share your notebook without having to set up a server to run the notebook for others to view. Is the GitHub team planning any kind of support for Marimo?

**Comment by akshayka** [comment, marked as answer]
Hi @ZackLarsen — we're working on this.

* We talked to GitHub and they suggested the best way forward would be for marimo to snapshot notebook outputs as ipynb or md (with outputs), as they can't render HTML
* We are currently working on exporting notebooks to ipynb from the command-line
* We will add an optional setting that automatically snapshots notebooks to ipynb in an auxiliary folder (`__marimo__`) while you're working on them
* We're open to doing the same for markdown in the future

In the meantime, you can export notebooks to HTML and send the HTML file to colleague (but it won't render on GitHub): https://docs.marimo.io/guides/exporting.html

Relevant issues to track:

* https://github.com/marimo-team/marimo/issues/2960
* https://github.com/marimo-team/marimo/issues/2961

**Reply by ZackLarsen** [reply]
Thanks for the update and for your work on this great tool!

**Reply by superlopuh** [reply]
@akshayka is there an issue to track the `__marimo__` idea? Would love a friendlier way to review marimo notebooks.

**Reply by akshayka** [reply]
It is already implemented and released, actually. You can enable it in the notebook settings menu (gear icon top right of the editor)

**Reply by superlopuh** [reply]
Amazing, thank you

**Reply by superlopuh** [reply]
Is there an issue that tracks export of jupyter notebooks with cells in topological sort order that would let all the cells be executed correctly?

**Reply by akshayka** [reply]
That exists in the command line. marimo export ipynb --help 

Do you want that in the automatic snapshots as well?

**Reply by superlopuh** [reply]
Our automated jupyter notebook tests fail as they can't execute the notebooks. Now that I think about it it's probably better for the sake of reviewing PRs to have the jupyter notebooks mirror the order of the marimo notebooks passed in. Disabling the jupyter tests in `__marimo__` folders should solve the problem.

**Reply by sstephenson-yelp** [reply]
@akshayka is there a plan to have output rendered directly in the Marimo py file at some stage, instead of converting to ipynb? I would love to be able to see diffs in both the notebook code and the results/output within one file. Is there a way to do this that I'm missing?
